Docs changes to reflect new assessment settings API and naming (#119)
hashicorp/terraform-docs-common: website/docs/cloud-docs/workspaces/settings/health-assessments.mdx

---
page_title: Health Assessments - Workspaces - Terraform Cloud
description: >-
  Enable automatic jobs to check for lifecycle concerns, like whether your real-world infrastructure matches
  your Terraform state file.
tfc_only: true
---

# Health Assessments

> -> **Note:** Health Asessments (formerly Drift detection) is in beta and is available in the paid **Business** tier. Refer to [Terraform Cloud pricing](https://www.hashicorp.com/products/terraform/pricing) for details and  [provide feedback](https://forms.gle/BwjgvrKQcLbgqK3f7) about how we can improve this feature.

Reviewer: First - will this still be in beta when you merge these docs? I don't think so, but you know, good to check

```suggestion
-> **Note:** Health Assessments are available in the [Terraform Cloud **Business** tier](https://www.hashicorp.com/products/terraform/pricing). Please [provide feedback](https://forms.gle/BwjgvrKQcLbgqK3f7) about how we can improve this feature.
```

> The term health assessment is used to describe the regular, automatic inspection of a workspace to surface resource lifecycle concerns. Presently, this includes infrastructure drift detection but may include other concerns in the future.
> 
> Infrastructure drift means that your real-world infrastructure no longer matches the configuration in your Terraform state file. Drift occurs when a user modifies resources outside of the Terraform workflow. For example, a colleague may update resource configuration directly in the cloud provider console during a production incident, changing the resource attributes from those in tracked in your state file.
> 
> > **Hands On:** Try our [Manage Resource Drift](https://learn.hashicorp.com/tutorials/terraform/resource-drift) tutorial on HashiCorp Learn to review how to detect and resolve drift.
> 
> Health assessments is a workspace-level setting. When enabled, Terraform Cloud will automatically run a health assessment for that workspace every 24 hours. The assessment checks the actual settings of your infrastructure against those tracked in your workspace's state file.

Reviewer: we want to stay away from future-facing statements like "we will add more in the future." I know it's a bit odd because drift detection is the only one, so I massaged it a bit --> thoughts?

```suggestion
Health assessment describes the regular, automatic inspection of a workspace to surface resource lifecycle concerns. Drift detection is a type of health assessment.

Infrastructure drift means that your real-world infrastructure no longer matches the configuration in your Terraform state file. Drift occurs when a user modifies resources outside of the Terraform workflow. For example, a colleague may update resource configuration directly in the cloud provider console during a production incident, changing the resource attributes from those in tracked in your state file.

> **Hands On:** Try our [Manage Resource Drift](https://learn.hashicorp.com/tutorials/terraform/resource-drift) tutorial on HashiCorp Learn to review how to detect and resolve drift.

When enabled, Terraform Cloud automatically runs a health assessment for your workspace every 24 hours. The assessment checks the actual settings of your infrastructure against those tracked in your workspace's state file.
```

## Requirements

Workspaces require the following settings to receive health assessments:
- Terraform version 0.15.4+
- [Remote execution mode](/cloud-docs/workspaces/settings#execution-mode) or [Agent execution mode](/cloud-docs/agents#configuring-workspaces-to-use-the-agent) for Terraform runs

## Enable Health Assessments

> To enable health assessments within a workspace:

Reviewer: ```suggestion
To enable health assessments within a specific workspace:
```

1. Verify that the workspace satisfies the [requirements](#requirements).
1. Go to the workspace and click **Settings > General**.
1. Select **Enable** under **Health Assessments**.
1. Click **Save settings**.

> > **Note:** You may also enforce assessments across all workspaces in an organization from the [organization assessments settings](/cloud-docs/users-teams-organizations/organizations/#assessments)

Reviewer: I'm thinking that this doesn't need to be a note --> it's a legitimate enablement method! I also wanted to add the tidbit here about how when you do that, it overrides whatever you did on the workspace because I think that's an important point!

```suggestion
You can also enforce health assessments for all eligible workspaces in an organization within the organization's **Assessments** settings. Enforcing health assessments at an organization-level overrides workspace-level settings. 
```

### Health Assessment Scheduling

If there are no active Terraform runs in the workspace, Terraform Cloud starts a health assessment as soon as you enable the feature. If you enable health assessments during a speculative plan, Terraform Cloud will start a health assessment soon after that plan's completion. If you enable health assessments during other types of runs, Terraform Cloud will start the first assessment in about 24 hours.

A health assessment never interrupts or interferes with runs. Terraform Cloud triggers a health assessment if at least 24 hours have passed since the last assessment and there are no active runs in the workspace.

You can start a new run during a health assessment, but it will cancel the current assessment. Terraform Cloud will run the next assessment in 24 hours.

> If health assessments are enabled on multiple workspaces, assessments may run concurrently. Health assessments do not affect your concurrency limit. Terraform Cloud also monitors and controls health assessment concurrency, so it will not cause issues for large-scale deployments with thousands of workspaces. However, Terraform Cloud performs health assessments in batches, so health assessments may take longer to complete when you enable it in a large number of workspaces.

Reviewer: ```suggestion
If health assessments are enabled on multiple workspaces, assessments may run concurrently. Health assessments do not affect your concurrency limit. Terraform Cloud also monitors and controls health assessment concurrency, so it will not cause issues for large-scale deployments with thousands of workspaces. However, Terraform Cloud performs health assessments in batches, so health assessments may take longer to complete when you enable them in a large number of workspaces.
```
